salt/cli/batch.py

```diff
@@ -26,6 +26,59 @@
 log = logging.getLogger(__name__)
 
 
+def _batch_get_opts(
+        tgt,
+        fun,
+        parent_opts,
+        arg=(),
+        tgt_type='glob',
+        ret='',
+        kwarg=None,
+        batch='10%',
+        **kwargs):
+    # We need to re-import salt.utils.args here
+    # even though it has already been imported.
+    # when cmd_batch is called via the NetAPI
+    # the module is unavailable.
+    import salt.utils.args
+
+    arg = salt.utils.args.condition_input(arg, kwarg)
+    opts = {'tgt': tgt,
+            'fun': fun,
+            'arg': arg,
+            'tgt_type': tgt_type,
+            'ret': ret,
+            'batch': batch,
+            'failhard': kwargs.get('failhard', False),
+            'raw': kwargs.get('raw', False)}
+
+    if 'timeout' in kwargs:
+        opts['timeout'] = kwargs['timeout']
+    if 'gather_job_timeout' in kwargs:
+        opts['gather_job_timeout'] = kwargs['gather_job_timeout']
+    if 'batch_wait' in kwargs:
+        opts['batch_wait'] = int(kwargs['batch_wait'])
```

Comment:
I am wondering why `batch_wait` needed to be `int()`'ed here. I'd rather add it to the config, same as `timeout` etc.

Reply:
And also shrink this too in `for opt...` loop, rather three time `if`.

Reply:
@isbm this could all be good ideas but  I don't want to incorporate them at this point because I'm trying to do a refactoring here. The refactoring has already a high chance of introducing you bugs and adding additional changes would raise it even more.
Let's try to keep all these ideas for after I am done with the refactoring and let's do them in the following iteration. Agreed?

Reply:
Ah, WP, right. Yeah, we can make it better with the follow-up then.